src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.go

```diff
@@ -100,19 +102,21 @@ func (st *StoragePowerActorState_I) _lockPledgeCollateral(rt Runtime, address ad
 	// AvailableBalance -> LockedPledgeCollateral
 	// TODO: potentially unnecessary check
 	if amount < 0 {
-		rt.Abort("negative amount.")
+		rt.AbortArgMsg("negative amount.")
 	}
 
 	// TODO: convert address to MinerActorID
 	var minerID addr.Address
 
 	currEntry, found := st.PowerTable()[minerID]
 	if !found {
-		rt.Abort("minerID not found.")
+		rt.AbortArgMsg("minerID not found.")
 	}
 
 	if currEntry.Impl().AvailableBalance() < amount {
-		rt.Abort("insufficient available balance.")
+		rt.Abort(
+			exitcode.UserDefinedError(exitcode.InsufficientFunds_User),
```

Comment:
Agreed. `UserDefinedError` may be the wrong term. The intent is "`NonSystemError`", i.e., an error that results from the logic of a particular actor, rather than an error in the VM/runtime itself.